scripts/babel/extension-resolver.js

```diff
@@ -0,0 +1,142 @@
+// original source: https://github.com/shimataro/babel-plugin-module-extension-resolver
+// To create proper ES Modules, the paths imported need to be fully-specified,
+// and can't be resolved like is possible with CommonJS. To have large compatibility
+// without much hassle, we don't use extensions *inside* the source code, but add
+// them with this plugin.
+
+const fs = require('fs');
+const path = require('path');
+
+const PLUGIN_NAME = 'babel-plugin-extension-resolver';
+
+const importStartsWith = ['instantsearch.js/', 'react-dom/'];
```

Comment:
Are those exceptions, or could we generalize this behavior?

Reply:
I'm not actually 100% sure why react-dom/server also is required even though it's CommonJS, but I had an error in Next if that wasn't the case. There's other dependency (dequal/lite) that errored if I *did* add the extension. I'll also investigate what it is that makes the module required to be fully resolved for react-dom

Reply:
These are the two dependencies that are required according to node. I think it's because dequal explicitly mentions it's cjs, but react doesn't

Reply:
That's possible. Perhaps we can add a comment explaining why these 2 deps need to be treated as such?

Then as long as the module test fails without an item in this array, that's good to me.